contrib/platform/src/com/sun/jna/platform/win32/COM/util/ProxyObject.java

```diff
@@ -353,6 +358,37 @@ public void unadvise(Class<?> comEventCallbackInterface, final IComEventCallback
     }
 
     // --------------------- IDispatch ------------------------------
+
+    @Override
+    public ITypeInfo getTypeInfo() {
+        com.sun.jna.platform.win32.COM.IDispatch mayBeDispatch = ProxyObject.this.getRawDispatch();
+
+        PointerByReference pDispatch = new PointerByReference();
+        HRESULT hr = mayBeDispatch.QueryInterface(new REFIID(IDispatch.IID_IDISPATCH), pDispatch);
+
+        if (!WinNT.S_OK.equals(hr)) {
```

Comment:
I think this should raise an exception.

Reply:
It is not guaranteed to have a IDispatch here, i did not raise an exception here for two reasons:

1. The running object table assumes all running object are IDispatch, this is not true for some and i had some Error's raised by the GetTypeInfoCount call. This might be a bug in the EnumMoniker, there is an open todo at line 113. I think changing this would break the public api.

2. It is possible to get into this state without doing something wrong like this:
``
// 1. the java friendly implementation of IDispatch, usually a ProxyObject instance
IDispatch dispatch = ?;

 // 2. get a ProxyObject with the ProxyObject#rawDispatch = IUnkown
IUnkown unkown = dispatch.queryInterface(IUnkown.class);

// 3. query for an interface with \\@ComInterface(name=...)
unkown.queryInterface(InterfaceByName.class); 

// 4. an Error is raised at 3.
``